### ISSUE TITLE
Masking mouse brains

### ISSUE THREAD

**Issue by araikes** [nipreps/nirodents#43]
Hi @oesteban and @eilidhmacnicol ,

I was curious what the status of nirodents is. I've been spending much of my day looking for a good masking strategy for some mouse data and I came across the OSF preprint for your brain extraction here. I'm just curious what would be necessary in order to make this usable for mouse data. I realize that currently it's using the Fischer344 rat atlas.

Thanks

**Comment by oesteban** [comment]
I would give it a quick try with the Fischer344 Atlas and see if it results are decent. If not, perhaps with the same template, but manipulating the affine to scale the data. If that doesn't work either, we would need a mouse template.

**Comment by araikes** [comment]
Sounds good. I'll give it a shot.

Thanks @oesteban

**Comment by araikes** [comment]
Yeah... out of the box running with no options set (e..g, `singularity run --cleanenv -B path/to/infolder:/inputfolder -B path/to/out:/output /path/to/nirodents.sif /infolder/img_T2w.nii.gz`) grabbed the Fisher template and ran but gave me a whole head mask.

![sub-control01_acq-iso_desc-preproc_T2w](https://user-images.githubusercontent.com/18663624/103708126-37bb8d80-4f6d-11eb-8357-7f3b403d529b.png)

**Comment by eilidhmacnicol** [comment]
Hi - @araikes 
I can't say I'm surprised that the Fischer344 template didn't work well. Adding the Allen brain atlas and/or the AMBMC atlas to [templateflow](https://www.templateflow.org) has been high on my list of priorities since we released the preprint, but I haven't had the bandwidth to get around to it. 

I would hope that species-specific templates would go some way to improve the results, but I don't know for sure if there would be any other parameters that would require tuning after swapping out the templates. Is there a particular mouse template that you would usually use?

**Comment by araikes** [comment]
Hi @eilidhmacnicol,

I just saw this. Sorry for the long lag. ABA/AMBMC would be good as targets. We frequently use the Dorr and DSURQE atlases (https://wiki.mouseimaging.ca/display/MICePub/Mouse+Brain+Atlases) as well.

Really just having something containerized, reproducible, and straightfoward would be tremendous.

**Comment by eilidhmacnicol** [comment]
Thanks for the feedback @araikes - I know colleagues in my lab use DSURQE. The problem is that the folks in Montreal have made the atlases available, but (AFAIK) the atlases are not distributed with a licence. This poses some problems for integration with templateflow, since without an explicit licence, we have to assume the most stringent restrictions on uses (all copyrights reserved).

Personally, I don't do much work with mouse data, so I have been a bit slow to extending it to mice, though I understand that there is the appetite for it (even within my lab). I have been talking to my mouse-focused colleagues about uploading the ABA, but if you (or others you may know) would like to be involved, I think we could progress faster. The support from @oesteban et al. has been excellent, but their experience with preclinical imaging is limited, so including more preclinical-focused contributors could help us push forward.

Otherwise, know that adding a mouse template is high on my to-do list!

**Comment by araikes** [comment]
I'd be happy to get involved... I'm relatively new to preclinical (doing it for about a year, and hacking it together as I go) but have a vested interest in getting containerized, useful workflows but I wouldn't even know where to begin. Maybe @oesteban can point me in the relevant direction. 

I'll reach out to the Montreal group and see what the licensing is for the Dorr/DSURQE atlases. I know their software (Pydpiper) has a pretty open use license for academic work.

**Comment by araikes** [comment]
@eilidhmacnicol and @oesteban,

Sorry to resurrect this after more than a year but I have opportunity to turn my attention back to it. I have email confirmation from two individuals at the Montreal group that the templates/masks/atlases that there are no restrictions on the inclusion of the templates/masks/atlases in other softwares. Is that sufficient for their inclusion with TemplateFlow and therefore use here?

**Comment by eilidhmacnicol** [comment]
Hey @araikes - I'm very glad to see that you're still interested.

This is good and bad news. The good news is that the only requirement for a template to be available via TemplateFlow is a permissive license. It sounds like the Montreal group are happy for the permissive part (which is most often the blocker), but unfortunately, email confirmation is not enough. We'd enthusiastically encourage the creation of a clear license file that is distributed with the templates themselves, which can be as little work as copying and pasting an appropriate plain text file from https://choosealicense.com

Without this, we (and other users) can't be certain what we can and can't do with the templates.

In other news, we have been working on adding another mouse template to TemplateFlow but has run into some issues (see: templateflow/templateflow#140), but @oesteban has assured me that this will be addressed urgently. TemplateFlow would certainly still benefit from the Dorr/DSURQE resources, so I think we should still encourage them to provide a license with the template, but I let you know this as an option in the mean time.

**Comment by araikes** [comment]
Hi @eilidhmacnicol,

I'll see if I can persuade them along the license rabbit hole. My guess is no but I understand the challenge.

I saw the mouse brain in TemplateFlow when I was thinking about this yesterday but it didn't look like things were ready. Beyond the issues noted in the issue linked, it looks like there are some other issues, like the `res-1` and `res-2` T1 maps are the same resolution and space (25 micron isotropic, same origin) while the `mask` and `probseg` images are different (150 micron isotropic, different origin from the T1 maps).

While Dorr/DSURQE would be beneficial, one other alternative would be https://www.nitrc.org/projects/tmbta_2019/. It's distributed with a license (non-commercial use but it's currently available through NITRC).

**Comment by eilidhmacnicol** [comment]
aha - I actually did look into the Turone mouse atlas, but the license that is distributed with seems to be for a behaviour tracking software. I reached out to the team behind the atlas for clarification and they didn't get back to me.
From the `license.txt` (downloaded 7th May 2022):
> Developer has developed and licenses to users its software program under the name MouseTracker (the "Software"). Licensee desires to utilize a copy of the Software. Now, therefore, Developer and Licensee agree as follows: 1. Developer hereby grants to Licensee a perpetual, non-exclusive, limited license to use the Software as set forth in this Agreement. 2. Licensee commits to cite this software in reports and publications, if used: Freeman, J.B. & Ambady, N. (2010). MouseTracker: Software for studying real-time mental processing using a computer mouse-tracking method. Behavior Research Methods, 42, 226-241.

Thanks for your perseverance on this issue. Please know that this has been an ongoing issue over the last year, and one of the primary reasons we developed the mousein template.

On that front, we have some good news. If you can use Datalad under the hood of templateflow (i.e., if you have access to a system that can run git annex, which seems to be linux/intel-based macs but not apple silicon and windows), then the MouseIn template is fixed. The problem seems to be with templateflow's S3 mirror (see: https://github.com/templateflow/templateflow/issues/140#issuecomment-1120158330)

**Comment by DavidBarriere** [comment]
Dear eilidhmacnicol,
As the manager of the TMBTA, I am happy to announced is completely free for use for academic only. Is our goal to create and share brain templates and atlases for MRI (SIGMA rat Template, Turone mouse template, Turone Sheep Template).
I hope that our tools will help you !

Best 
D

**Comment by eilidhmacnicol** [comment]
@DavidBarriere this is great news! We would be happy to redistribute them through TemplateFlow (with the necessary caveats that it is for academic use only).

Can you provide datasets packaged with unambiguous versions of the license? Then either we can prepare it and upload it for you or we can help you to upload it.

**Comment by DavidBarriere** [comment]
We could discuss deeper about this.
My email : david.barriere@cnrs.fr
